source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RendererUtil.java

```diff
@@ -148,13 +163,121 @@ public void handleTag(boolean opening, String tag, Editable output, XMLReader xm
                 {
                     output.append("\n• ");
                 }
+                tagWasHandled = true;
             }
 
             if (tag.equals("ol") && opening)
             {
                 orderedList = true;
-                tagNumber = 1;
+                String tagNumberString = getAttribute("start", attributes);
+
+                int retrievedTagNumber = 1;
+                if (tagNumberString != null)
+                {
+                    retrievedTagNumber = Integer.parseInt(tagNumberString);
+                }
+
+                tagNumber = retrievedTagNumber;
+                tagWasHandled = true;
             }
+
+            return tagWasHandled;
+        }
+
+        @Override
+        public void handleTag(boolean opening, String tag, Editable output, XMLReader xmlReader)
+        {
+            if (defaultContentHandler == null)
+            {
+                // save input text
+                text = output;
+
+                // store default XMLReader object
+                defaultContentHandler = xmlReader.getContentHandler();
+
+                // replace content handler with our own that forwards to calls to default when needed
+                xmlReader.setContentHandler(this);
+
+                // handle endElement() callback for <inject/> tag
+                tagStatusQueue.addLast(Boolean.FALSE);
+            }
+        }
+
+        // ContentHandler override methods
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException
+        {
+            boolean isHandled = handleTag(true, localName, text, attributes);
+            tagStatusQueue.addLast(isHandled);
+
+            if (!isHandled)
+            {
+                defaultContentHandler.startElement(uri, localName, qName, attributes);
```

Comment:
I'm a little concerned here. what happens when the tag is not handled inside the `startElement`? Would it create an infinite loop? since it will call the method with the same arguments?

Reply:
we're handling a small subset of the html tags, if it's not handled instead of handling it ourselves we're passing the tag to the original/default content handler that we have been using all this time hence no infinite loop should ever happen